### PR TITLE
various atp and ssl JRF database refactoring

### PR DESCRIPTION
**Pull request by jshum2479**
This fixes -run_rcu against SSL database and various refactoring of both ATP and SSL JRF database domain creation.

### CODE REVIEW

**Review by ddsharpe — APPROVED** [review]

File: core/src/main/python/wlsdeploy/tool/create/domain_creator.py [hunk 1]
```diff
@@ -276,54 +276,66 @@ def __run_rcu(self):
...
+
+
+            self._set_rcu_ssl_args_properties(ssl_conn_properties, rcu_db_info, keystore, keystore_type, truststore,
+                                              truststore_type)
+
+            # hard coding for now, may need to expose it if ATP access changed later
+
+            if not ssl_conn_properties.has_key(DRIVER_PARAMS_NET_FAN_ENABLED):
+                ssl_conn_properties[DRIVER_PARAMS_NET_FAN_ENABLED] = 'false'
+            ssl_conn_properties[DRIVER_PARAMS_NET_SSL_VERSION] = '1.2'
```
Comment (robertpatrick) [inline core/src/main/python/wlsdeploy/tool/create/domain_creator.py:303]: why don't we set this only if it is not set, like you are doing for the previous property?

File: core/src/main/python/wlsdeploy/tool/create/domain_creator.py [hunk 2]
```diff
@@ -276,54 +276,66 @@ def __run_rcu(self):
...
+            ssl_conn_properties[DRIVER_PARAMS_NET_SERVER_DN_MATCH_PROPERTY] = 'false'
```
Comment (robertpatrick) [inline core/src/main/python/wlsdeploy/tool/create/domain_creator.py:304]: same comment about only setting if not already specified.

**Review by rakillen — APPROVED** [review]

File: core/src/main/python/wlsdeploy/tool/create/domain_creator.py (hunk 1 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/create/domain_creator.py:301]: Actually, the fan enabled should always be false, plus these properties must be that value for it to work properly.    There is comment in the code, not sure why it didn't show up in PR.

File: core/src/main/java/oracle/weblogic/deploy/create/RCURunner.java [hunk 3]
```diff
@@ -136,36 +135,23 @@ public static RCURunner createRunner(String domainType, String oracleHome, Strin
...
-        addExtraSSLPropertyFromMap(runnerMap, connectionProperties, sslArgs, "javax.net.ssl.keyStorePassword");
-        addExtraSSLPropertyFromMap(runnerMap, connectionProperties, sslArgs, "javax.net.ssl.trustStorePassword");
+        StringBuilder sslArgs = getSSLArgsStringBuilder(sslConnectionProperties);
 
+        addExtraSSLPropertyFromMap(runnerMap, sslConnectionProperties, sslArgs, "javax.net.ssl.keyStorePassword");
+        addExtraSSLPropertyFromMap(runnerMap, sslConnectionProperties, sslArgs, "javax.net.ssl.trustStorePassword");
 
         runner.atpDB = true; // "ATP".equals(databaseType);  // or scan if there are any 'ssl' in properties ?
-        runner.atpSSlArgs = sslArgs.toString();
+        runner.sSlArgs = sslArgs.toString();
```
[the remark this answers is not in the document]
Comment (jshum2479) [reply core/src/main/java/oracle/weblogic/deploy/create/RCURunner.java]: Fixed

File: core/src/main/java/oracle/weblogic/deploy/create/RCURunner.java [hunk 4]
```diff
@@ -199,44 +200,16 @@ private static void addExtraSSLPropertyFromMap(PyDictionary runnerMap, PyDiction
...
-        sslArgs.append(",javax.net.ssl.keyStoreType=" + keyStoreType);
-        if (keyStorePassword != null && !keyStorePassword.equals("None")) {
-            sslArgs.append(",javax.net.ssl.keyStorePassword="+ keyStorePassword);
-        }
-        sslArgs.append(",oracle.net.ssl_server_dn_match="+ matchType);
+        StringBuilder sslArgs = getSSLArgsStringBuilder(sslConnectionProperties);
 
         runner.sslDB = true;
-        runner.atpSSlArgs = sslArgs.toString();
+        runner.sSlArgs = sslArgs.toString();
```
[the remark this answers is not in the document]
Comment (jshum2479) [reply core/src/main/java/oracle/weblogic/deploy/create/RCURunner.java]: Fixed

**Review by robertpatrick — APPROVED** [review]

**Pull request merged**